### ISSUE TITLE
Why it is fine to normalize train, validation and test set independently in the ImageDataGenerator?

### ISSUE THREAD

**Issue by yxchng** [fchollet/keras#3245]
From what I understand from the code, you are normalizing the train set independently from the validation or test set. Isn't that wrong? Ideally shouldn't you normalize using the statistics of the train set to normalize the validation or test set? This is quite impossible in my opinion since the transformation is random. 

I am just wondering if that will mess up the performance of the classifier? Is there a theoretical justification for this?
